cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -489,28 +476,29 @@ pub enum SearchError {
 
 #[derive(Debug, Error)]
 pub enum ResolveError {
-    #[error("resolution failed")]
-    Resolve(#[source] APIError<api_types::ErrorResponse>),
+    #[error("resolution failed: {}", fmt_info(_0))]
+    Resolve(ResponseValue<api_types::ErrorResponse>),
     #[error(transparent)]
     CatalogClientError(#[from] CatalogClientError),
 }
-
 #[derive(Debug, Error)]
 pub enum VersionsError {
-    // TODO: would be nicer if this contained a ResponseValue<api_types::ErrorResponse>,
-    // but that doesn't implement the necessary traits.
-    #[error("getting package versions failed")]
-    Versions(#[source] APIError<api_types::ErrorResponse>),
+    #[error("getting package versions failed: {}", fmt_info(_0))]
+    Versions(ResponseValue<api_types::ErrorResponse>),
     #[error(transparent)]
     CatalogClientError(#[from] CatalogClientError),
 }
 
-impl CatalogClientError {
-    /// UnexpectedError corresponds to any variant of APIError other than
-    /// ErrorResponse, which is the only error that is in the API schema.
-    fn is_unexpected_error(error: &APIError<api_types::ErrorResponse>) -> bool {
-        !matches!(error, APIError::ErrorResponse(_))
-    }
+/// TODO: I copied this from the fmt_info function used by the Display impl of
+/// APIError.
+/// We should find something cleaner.
+fn fmt_info(error_response: &ResponseValue<api_types::ErrorResponse>) -> String {
+    format!(
+        "status: {}; headers: {:?}; value: {:?}",
+        error_response.status(),
+        error_response.headers(),
+        error_response.as_ref()
+    )
 }
```

Comment:
The issue is that `ResponseValue<api_types::ErrorResponse>` doesnt implement Error or Display?
Is the `Debug` output way too jarring?

Reply:
Yeah. `Debug` doesn't output `status` or `headers`